app/scripts/lib/message-manager.js

```diff
@@ -3,8 +3,45 @@ const ObservableStore = require('obs-store')
 const ethUtil = require('ethereumjs-util')
 const createId = require('./random-id')
 
+var msgData = {
+  id: msgId,
+  msgParams: msgParams,
+  time: time,
+  status: 'unapproved',
+  type: 'eth_sign',
+}
+
+/**
+ * Represents, and contains data about, an 'eth_sign' type signature request. These are created when a signature for
+ * an eth_sign call is requested.
+ *
+ * @see {@link https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign}
+ *
+ * @typedef {Object} Message
+ * @property {number} id An id to track and identify the message object
+ * @property {object} msgParams The parameters to pass to the eth_sign method once the signature request is approved.
+ * @property {object} msgParams.metamaskId Added to msgParams for tracking and identification within MetaMask.
+ * @property {string} msgParams.data A hex string conversion of the raw buffer data of the signature request
+ * @property {number} time The epoch time at which the this message was created
+ * @property {string} status Indicates whether the signature request is 'unapproved', 'approved', 'signed' or 'rejected'
+ * @property {string} type The json-prc signing method for which a signature request has been made. A 'Message' with
+ * always have a 'eth_sign' type.
+ *
+ */
 
 module.exports = class MessageManager extends EventEmitter {
+
+  /**
+   * Controller in charge of managing - storing, adding, removing, updating - Messages.
+   *
+   * @typedef {Object} MessageManager
+   * @param {object} opts @deprecated
```

Comment:
I know that JSDoc is technically case-insensitive when it comes to types, but I think we should stay consistent throughout (`object` vs. `Object`.)